### PR TITLE
Prefect pipeline

### LINKED ISSUE #44: Orchestrate for multiple files
Run get_bitinformation and get_keepbits once and apply bitround to multiple files. 

How to run in a script in parallel?
try prefect https://docs.prefect.io/core/concepts/persistence.html#output-caching-based-on-a-file-target pangeo_forge_recipes/executors/prefect.py

### PR DESCRIPTION
**Pull request by aaronspring**
Closes #44 

- adds `bp.get_prefect_flow(paths)`, which can be executed serially or in parallel with `dask`
- `DaskExecutor`s work on my laptop without any problem. on GHA and supercomputer I get segfaults.

### CODE REVIEW

File: tests/test_bitinformation_pipeline.py [hunk 1]
```diff
@@ -24,3 +25,55 @@ def test_full():
...
+        if os.path.exists(p):
+            os.remove(p)
+        if os.path.exists(p.replace(".nc", "_bitrounded_compressed.nc")):
+            os.remove(p.replace(".nc", "_bitrounded_compressed.nc"))
+
+
+@pytest.mark.parametrize("executor", ["local", "dask"])
+def test_get_prefect_flow_executor(flow_paths, executor):
+    """Test get_prefect_flow runs for different executors."""
+    flow, paths = flow_paths
```
Comment (aaronspring) [inline tests/test_bitinformation_pipeline.py:56]: assert paths not exist

File: bitinformation_pipeline/bitinformation_pipeline.py [hunk 2]
```diff
@@ -405,6 +405,169 @@ def plot_bitinformation(bitinfo):
...
+
+    Create prefect.Flow and run sequentially
+    >>> flow = bp.get_prefect_flow(paths=paths)
+    >>> import prefect
+    >>> logger = prefect.context.get("logger")
+    >>> logger.setLevel("ERROR")
+    >>> flow.run()
+    <Success: "All reference tasks succeeded.">
+
+    Inspect flow state
```
Comment (aaronspring) [inline bitinformation_pipeline/bitinformation_pipeline.py:456]: Combine

File: bitinformation_pipeline/bitinformation_pipeline.py [hunk 3]
```diff
@@ -405,6 +405,169 @@ def plot_bitinformation(bitinfo):
...
+    >>> st = flow.run()
+    >>> # requires graphviz
+    >>> # flow.visualize(st)
+
+    Run in parallel with dask:
+    >>> # import os  # https://docs.xarray.dev/en/stable/user-guide/dask.html
+    >>> # os.environ["HDF5_USE_FILE_LOCKING"] = "FALSE"
+    >>> from prefect.executors import LocalDaskExecutor
+    >>> executor = LocalDaskExecutor(scheduler="processes")
```
Comment (aaronspring) [inline bitinformation_pipeline/bitinformation_pipeline.py:465]: Rm outputs before to rerun

File: bitinformation_pipeline/bitinformation_pipeline.py [hunk 4]
```diff
@@ -405,6 +405,186 @@ def plot_bitinformation(bitinfo):
...
+    3. Apply bitrounding with `xr_bitround`
+    4. Save as compressed netcdf with `to_compressed_netcdf`
+
+    Many parameters can be changed when running the flow `flow.run(parameters=dict(chunk="auto"))`:
+    - paths: list of Paths
+        paths to be bitrounded
+    - analyse_paths: str or int
+        which paths to be passed to `bp.get_bitinformation`. choose from ["first_last", "all", int], where int is interpreted as stride, i.e. paths[::stride]. Defaults to "first".
+    - enforce_dtype : str or None
+        Enforce dype for all variables. Currently `get_bitinformation` fails for different dtypes in variables. Do nothing if None. Defaults to None.
```
Comment (aaronspring) [inline bitinformation_pipeline/bitinformation_pipeline.py:423]: `flow.run(parameters=dict(enforce_dtype="float32"))` circumvents error for files with vertices as data_vars as float64 when other vars are float32, as in ICON or MPIOM.

**Review by observingClouds — APPROVED** [review]
Feel free to merge.

**Pull request merged**